### DISCUSSION TITLE
Script Editor: Improve readability of GDScript functions, for-loops, ifs, etc.

### DISCUSSION THREAD

**Discussion by SirManfred** [godotengine/godot-proposals#9975] [Editor]
Since GDScript doesn't have the `{ }` wrappers to show where a function, for-loop or if, etc. begins and ends, something like this would be a nice option to have available. If not by default, then at least as an option in the Editor settings.
![image](https://github.com/godotengine/godot-proposals/assets/7838390/b8be676d-44c3-4974-9c99-0d5a4bb0c6d0)

I was also made aware that this might conflict with the `Draw Spaces`-option in the Editor Settings so I made a quick and dirty version for that as well in order to illustrate that it's possible to get around it, although it is not as neat as the previous version:
![image](https://github.com/godotengine/godot-proposals/assets/7838390/64d48ed8-1178-4d8c-beb4-137874d7e63a)

Alternatively, something like this could be an idea:
![image](https://github.com/godotengine/godot-proposals/assets/7838390/ccf527b2-835b-42bd-85d9-3df2c623273c)
A bit cleaner, doesn't clutter. Shows you the indented sections and the number of spaces or tabs used for each of them.

Anyways, I'd love to hear your thoughts about these!
Cheers!
/Frederik

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/8592.
